### PR TITLE
cleanup and more comments. 

### PR DESCRIPTION
**Pull request by vermas2012**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

**Review by sushanb — APPROVED** [review]

File: bigtable-hbase-replication/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicator.java [hunk 1]
```diff
@@ -243,16 +227,18 @@ public boolean replicate(Map<String, List<BigtableWALEntry>> walEntriesByTable)
...
+        // TODO handle the case where a single row has >100K mutations (very rare, but should not fail)
         numCellsInBatch += rowCells.getValue().size();
         batchSizeInBytes += getRowSize(rowCells);
         batchToReplicate.put(rowCells.getKey(), rowCells.getValue());
 
-        if (batchSizeInBytes >= batchSizeThresholdInBytes) {
-          LOG.debug("Replicating a batch of " + batchToReplicate.size() + " rows and "
+        // TODO add tests for batch split on size and cell counts
+        if (batchSizeInBytes >= batchSizeThresholdInBytes || numCellsInBatch >= 100_000-1) {
+          LOG.trace("Replicating a batch of " + batchToReplicate.size() + " rows and "
```
[the remark this answers is not in the document]
Comment (vermas2012) [reply bigtable-hbase-replication/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicator.java:238]: trace is a level below debug. there is no difference just relative prefernces. I can enable debug layer and trace will not show up.

File: bigtable-hbase-replication/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTask.java [hunk 2]
```diff
@@ -41,24 +40,24 @@
   @VisibleForTesting
   static class PutMutationBuilder implements MutationBuilder {
 
-    // TODO make it final
-    private Put put;
+    final private Put put;
+    boolean closed = false;
```
[the remark this answers is not in the document]
Comment (vermas2012) [reply bigtable-hbase-replication/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTask.java:44]: what do you have in mind?

[the remark this answers is not in the document]
Comment (sushanb) [reply bigtable-hbase-replication/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTask.java:44]: Given that the if statement never executes, i was thinking nonExistent or something.

[the remark this answers is not in the document]
Comment (vermas2012) [reply bigtable-hbase-replication/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTask.java:44]: discussed offline.

**Pull request merged**